RestructuredText/reStructuredText.sublime-syntax

```diff
@@ -1,220 +1,1202 @@
 %YAML 1.2
 ---
-# http://www.sublimetext.com/docs/3/syntax.html
-name: reStructuredText
-comment: syntax highlighting for reStructuredText http://docutils.sourceforge.net, based on rst mode from jEdit
+# https://docutils.sourceforge.io/docs/user/rst/quickref.html
+# https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html
+# https://docutils.sourceforge.io/docs/ref/rst/directives.html
+# https://docutils.sourceforge.io/docs/ref/doctree.html
+# http://www.sublimetext.com/docs/syntax.html
+
+# NOTE: This syntax definition makes heavy use of backreferences.
+#       You might want to read the docs to make yourself familiar.
+
+name: reStructuredText (reST)
+scope: text.restructuredtext
+version: 2
+
 file_extensions:
   - rst
   - rest
-scope: text.restructuredtext
+
+###[ CONTEXTS ]################################################################
+
 contexts:
+
+  prototype:
+    # NOTE: Comments are not part of prototype, but regular explicit markup
+    #       blocks, i.e. body elements.
+    - include: merge-conflict-markers
+
   main:
-    - match: '^(?=(\s*)\S)'
-      push:
-        - meta_content_scope: meta.paragraph.restructuredtext
-        - match: ^(?!\1(?=\S))
-          pop: true
-        - include: inline
-  inline:
-    - match: '^(\s*)((\.\.)\s+raw(::)) html'
-      comment: directives.html
-      captures:
-        2: meta.directive.restructuredtext
+    - include: restructuredtext
+
+###[ RESTRUCTUREDTEXT ]########################################################
+
+  restructuredtext:
+    - include: between-empty-lines
+    - include: paragraphs
+
+  paragraphs:
+    # https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#paragraphs
+
+    - meta_include_prototype: false
+    - match: ^(?=([[:blank:]]*)\S)
+      push: paragraph-body
+
+  paragraph-body:
+    - meta_content_scope: meta.paragraph.restructuredtext
+    - match: '^(?!\1(?=\S))'
+      pop: 1
+    - include: body-elements
+    - include: inline-markup
+
+  between-empty-lines:
+    - match: ^$\n
```

Comment:
If linefeed is to be consumed and it is sure it not being already consumed by possible interpolating template rules, `$\n` is redundant. Either use `$` (non-consuming, matches end of file as well) or `\n` consuming newlines only.

Hence the following would be enough, assuming empty lines are not to be matched by other contexts on same level.

```suggestion
    - match: ^\n
```